appengine/expiration.py

```diff
@@ -0,0 +1,49 @@
+"""
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+"""Delete expired XML.
+"""
+
+__author__ = "fenichel@google.com (Rachel Fenichel)"
+
+
+import storage
+import datetime
+
+from google.cloud import ndb
+
+def delete_expired():
+  # Delete entries that have not been accessed in more than a year.
+  EXPIRATION_DAYS = 365
+  # Limit the query to avoid timeouts.
+  QUERY_LIMIT = 1000
+  bestBefore = datetime.datetime.utcnow() - datetime.timedelta(days=EXPIRATION_DAYS)
+  client = ndb.Client()
+  with client.context():
+    query = storage.Xml.query(storage.Xml.last_accessed < bestBefore)
+    results = query.fetch(limit=QUERY_LIMIT, keys_only=True)
+    for x in results:
+      x.delete()
```

Comment:
Purely a suggestion, that I don't know if we want.
We could add logging here, which would give us an idea of how often/when things are deleted.
```
import logging
...
logging.info('Cleaned up entry with key :%s', x.urlsafe())
```

Reply:
I don't think it's worth the logging here either. I expect that this cron job will delete a lot of entries at the beginning, and then we'll take a look at the number of remaining entities and make decisions from there.